Doc/reference/datamodel.rst

```diff
@@ -1614,30 +1614,25 @@ instances cannot override the behavior of a property.
 __slots__
 ^^^^^^^^^
 
-By default, instances of classes have a dictionary for attribute storage.  This
-wastes space for objects having very few instance variables.  The space
-consumption can become acute when creating large numbers of instances.
-
-The default can be overridden by defining *__slots__* in a class definition.
-The *__slots__* declaration takes a sequence of instance variables and reserves
-just enough space in each instance to hold a value for each variable.  Space is
-saved because *__dict__* is not created for each instance.
+*__slots__* allow us to explicitly declare data members (like
+properties) and deny the creation of *__dict__* and *__weakref__*
+(unless explicitly declared in *__slots__* or available in a parent.)
 
+The space saved over using *__dict__* can be significant.
 
 .. data:: object.__slots__
 
-   This class variable can be assigned a string, iterable, or sequence of
-   strings with variable names used by instances.  *__slots__* reserves space
-   for the declared variables and prevents the automatic creation of *__dict__*
-   and *__weakref__* for each instance.
+   This class variable can be assigned a string, iterable, sequence of
+   strings, or mapping with variable names used by instances.
```

Comment:
I would suggest leaving out *mapping*.  That possibility is covered by *iterable* without suggesting that a mapping would be a good thing to do (because order isn't guaranteed, because the values aren't used, and because it doesn't make much sense semantically).